changelog.d/pr-7410.md

```diff
@@ -0,0 +1,3 @@
+### 🚀 Enhancements and New Features
+
+- BF: gitlab hierarchy fix.  [PR #7410](https://github.com/datalad/datalad/pull/7410) (by [@jsheunis](https://github.com/jsheunis))
```

Comment:
```suggestion
- BF: Make the behavior of ``create-sibling-gitlab``'s ``--layout hierarchy`` parameter match its documentation. Previously, despite the documentation saying otherwise, the toplevel dataset in a recursive ``create-sibling-gitlab`` call became a GitLab project instead of being placed in a dedicated group. Now, ``--layout hierarchy`` creates a group per dataset, and places the dataset as a project inside this group, as the docstring promises.  [PR #7410](https://github.com/datalad/datalad/pull/7410) (by [@jsheunis](https://github.com/jsheunis))
```